pkg/server/targets/remove.go

```diff
@@ -34,6 +34,12 @@ func (s *TargetService) RemoveTarget(ctx context.Context, targetId string) error
 		return s.handleRemoveError(ctx, t, err)
 	}
 
+	err = s.revokeApiKey(ctx, targetId)
```

Comment:
Let's actually fail the entire operation here.

We have forced removal that can ignore this but for non-force, it can fail